deployer/deployer-services/src/main/java/io/hyscale/deployer/services/deployer/impl/KubernetesDeployer.java

```diff
@@ -427,4 +434,89 @@ private void deltaWait(long milliSeconds) {
         } catch (InterruptedException e) {
         }
     }
+
+    @Override
+    public List<ReplicaInfo> getReplicas(AuthConfig authConfig, String appName, String serviceName, String namespace, 
+            boolean isFilter) throws HyscaleException {
+        V1PodHandler v1PodHandler = (V1PodHandler) ResourceHandlers.getHandlerOf(ResourceKind.POD.getKind());
+        String selector = ResourceSelectorUtil.getServiceSelector(appName, serviceName);
+        ApiClient apiClient = clientProvider.get((K8sAuthorisation) authConfig);
+        List<V1Pod> podList = v1PodHandler.getBySelector(apiClient, selector, true, namespace);
+        
+        if (podList == null || podList.isEmpty()) {
+            return null;
+        }
+        
+        if (!isFilter) {
+            return K8sReplicaUtil.getReplicaInfo(podList);
+        }
+        
+        if (!PodPredicates.isPodAmbiguous().test(podList)) {
+            return K8sReplicaUtil.getReplicaInfo(podList);
+        }
+        String podOwner = K8sPodUtil.getPodsOwner(podList);
+        ResourceKind podOwnerKind = ResourceKind.fromString(podOwner);
+        
+        // Unknown parent
+        if (StringUtils.isBlank(podOwner)) {
+            logger.debug("Unable to determine latest deployment, displaying all replicas");
+            WorkflowLogger.warn(DeployerActivity.LATEST_DEPLOYMENT_NOT_IDENTIFIABLE);
+            return K8sReplicaUtil.getReplicaInfo(podList);
+        }
+        
+        // Deployment
+        if (ResourceKind.REPLICA_SET.equals(podOwnerKind) || ResourceKind.DEPLOYMENT.equals(podOwnerKind)) {
+            // Get deployment, get revision, get RS with the revision, get all labels and filter pods
+            return K8sReplicaUtil.getReplicaInfo(getDeploymentFilteredPods(apiClient, appName, serviceName, namespace, podList));
+        }
+        
+        // TODO do we need to handle STS cases ??
+        logger.debug("Replicas info:: unhandled case, pod owner: {}", podOwner);
+        return K8sReplicaUtil.getReplicaInfo(podList);
+    }
+    
+    private List<V1Pod> getDeploymentFilteredPods(ApiClient apiClient, String appName, String serviceName, String namespace, List<V1Pod> podList){
```

Comment:
This should be exposed as a service. Will be very helpful even for status also.

Reply:
This method is complicated to read, so better add some comments.

Reply:
Also , it would be great if we split this into multiple methods based on responsibility